pkg/client/client.go

```diff
@@ -18,15 +20,29 @@ import (
 var sessions = awsds.NewSessionCache()
 
 // CreateXrayClient creates a new session and xray client and sets tracking header on that client
-func CreateXrayClient(region string, datasourceInfo *awsds.AWSDatasourceSettings) (*xray.XRay, error) {
-	sess, err := sessions.GetSession(region, *datasourceInfo)
+func CreateXrayClient(awsSettings *awsds.AWSDatasourceSettings, backendSettings *backend.DataSourceInstanceSettings) (*xray.XRay, error) {
+	httpClientProvider := httpclient.NewProvider()
+	httpClientOptions, err := backendSettings.HTTPClientOptions()
+	if err != nil {
+		return nil, err
+	}
+	httpClient, err := httpClientProvider.New(httpClientOptions)
+	if err != nil {
+		backend.Logger.Error("failed to create HTTP client", "error", err.Error())
+		return nil, err
+	}
+	sess, err := sessions.GetSession(awsds.SessionConfig{
+		Settings:      *awsSettings,
+		HTTPClient:    httpClient,
+		UserAgentName: aws.String("X-ray"),
```

Comment:
the user agent is set a bit later (line 49), you need to remove that. Also, we should add something related to that in the changelog since it will change. See: https://github.com/grafana/timestream-datasource/blob/main/CHANGELOG.md#tbd

Reply:
I won't be able to log the user-agent here anymore, will I?

Reply:
you can log it with `sessions.GetUserAgentString("X-ray")` if  you want to